pkgs/development/python-modules/tinygrad/fix-dlopen-cuda.patch

```diff
@@ -1,32 +1,49 @@
 diff --git a/tinygrad/runtime/autogen/cuda.py b/tinygrad/runtime/autogen/cuda.py
-index 359083a9..3cd5f7be 100644
+index a30c8f53..e2078ff6 100644
 --- a/tinygrad/runtime/autogen/cuda.py
 +++ b/tinygrad/runtime/autogen/cuda.py
-@@ -143,10 +143,25 @@ def char_pointer_cast(string, encoding='utf-8'):
-     return ctypes.cast(string, ctypes.POINTER(ctypes.c_char))
+@@ -145,7 +145,19 @@ def char_pointer_cast(string, encoding='utf-8'):
  
  
-+NAME_TO_PATHS = {
-+    "libcuda.so": ["@driverLink@/lib/libcuda.so"],
-+    "libnvrtc.so": ["@libnvrtc@"],
-+}
-+def _try_dlopen(name):
-+    try:
-+        return ctypes.CDLL(name)
-+    except OSError:
-+        pass
-+    for candidate in NAME_TO_PATHS.get(name, []):
-+        try:
-+            return ctypes.CDLL(candidate)
-+        except OSError:
-+	        pass
-+    raise RuntimeError(f"{name} not found")
- 
  _libraries = {}
 -_libraries['libcuda.so'] = ctypes.CDLL(ctypes.util.find_library('cuda'))
--_libraries['libnvrtc.so'] = ctypes.CDLL(ctypes.util.find_library('nvrtc'))
-+_libraries['libcuda.so'] = _try_dlopen('libcuda.so')
-+_libraries['libnvrtc.so'] = _try_dlopen('libnvrtc.so')
++libcuda = None
```

Comment:
This just unfolds the `_try_dlopen` helper?

Reply:
Yes, I removed the function because there are no more 2 libs now.

Reply:
Would you be interested in having that @wozeparrot ? If so I can make a PR.

Reply:
Yea, should be fine to pr, would need to change in the `autogen_stubs.sh` file. We have a `try_dlopen` thing in autogen but its only used for amd_comgr right now: https://github.com/tinygrad/tinygrad/blob/5a7cb8d5a55bb4f5dc2dcf6dd56cd6bfd3b8f76c/autogen_stubs.sh#L25